pages/developers/tangle-blueprint.mdx

```diff
@@ -0,0 +1,261 @@
+use gadget_sdk::request_hook;
+
+#[request_hook(evm = "RequestContract")]
+pub fn on_request(nft_id: u64) {
+// Handle request logic here
+}
+
+### 1.8 Add Benchmarks
+
+Benchmarks help ensure your AVS performs efficiently. Here's how to define a benchmark:
+
+```rust
+use gadget_sdk::benchmark;
+
+#[benchmark(job_id = 0, cores = 2)]
+fn keygen_2_of_3() {
+    let n = 3;
+    let t = 2;
+    let result = keygen(&MyContext, n, t);
+    assert!(result.is_ok());
+}
+```
+
+This benchmark tests the keygen job with specific parameters.
+
+### 1.9 Error Handling
+
+Define custom error types for your AVS to handle various failure scenarios:
+
+```rust
+#[derive(Debug, Clone, Copy)]
+pub enum Error {
+    InvalidKeygen,
+    InvalidSignature,
+    InvalidRefresh,
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        let msg = match self {
+            Error::InvalidKeygen => "Invalid Keygen",
+            Error::InvalidSignature => "Invalid Signature",
+            Error::InvalidRefresh => "Invalid Refresh",
+        };
+        write!(f, "{}", msg)
+    }
+}
+
+impl std::error::Error for Error {}
+```
+
+By following these steps, you'll have a well-structured Tangle Blueprint AVS with defined jobs, reports, hooks, and benchmarks. In the next section, we'll cover how to test and deploy your AVS to the Tangle Network.
+
+## 2. Implementing the AVS Client (main.rs)
+
+Now that we've defined the core functionality of our AVS, let's implement the client that will handle registration, running, and event listening processes.
+
+### 2.1 Setting up the Main File
+
+Create a new file named `main.rs` in your project's `src` directory. This file will serve as the entry point for your AVS client. If you used the Tangle CLI to create your Blueprint this should already exist.
+
+### 2.2 Imports and Struct Definition
+
+Start by adding the necessary imports and defining the `TangleGadgetRunner` struct:
+
+```rust
+use color_eyre::{eyre::eyre, Result};
+use gadget_sdk::config::{ContextConfig, GadgetCLICoreSettings, GadgetConfiguration, StdGadgetConfiguration};
+use gadget_sdk::{
+    config::Protocol,
+    events_watcher::{substrate::SubstrateEventWatcher, tangle::TangleEventsWatcher},
+    tangle_subxt::tangle_testnet_runtime::api,
+    tx,
+    info
+};
+use std::io::Write;
+use crypto_service_blueprint as blueprint;
+use structopt::StructOpt;
+use gadget_sdk::keystore::KeystoreUriSanitizer;
+use gadget_sdk::keystore::sp_core_subxt::Pair;
+use gadget_sdk::run::GadgetRunner;
+use gadget_sdk::tangle_subxt::subxt::tx::Signer;
+
+struct TangleGadgetRunner {
+    env: GadgetConfiguration<parking_lot::RawRwLock>,
+}
+```
+
+### 2.3 Implementing the GadgetRunner Trait
+
+Implement the `GadgetRunner` trait for `TangleGadgetRunner`:
+
+```rust
+#[async_trait::async_trait]
+impl GadgetRunner for TangleGadgetRunner {
+    type Error = color_eyre::eyre::Report;
```

Comment:
Shouldn't this use the custom error from 1.9 instead of assuming `color_eyre` is available? Same goes for the use of `color_eyre::Result`.

Reply:
It's auto-pulled from some examples. Can you help fix it up to your liking? I support and want everyone to be contributing to changes here.